src/is-my-room/index.ts

```diff
@@ -0,0 +1,25 @@
+import {isRoomVisible} from '@open-screeps/is-room-visible'
+
+export function isMyRoom(roomName: string | Room): boolean{
+  if(!roomName){
+    return false
+  }
+
+  let room: Room
+
+  if(typeof roomName === 'string'){
+    if(!isRoomVisible(roomName)){
+      return false
+    }
+
+    room = Game.rooms[roomName]
+  }else{
+    room = roomName
```

Comment:
I prefer recursion over this approach, but it obviously works like this as well. :)